src/main/java/com/treasuredata/client/TDHttpClient.java

```diff
@@ -408,15 +410,38 @@ else if (e.getCause() instanceof SSLException) {
         throw rootCause.get();
     }
 
-    protected TDClientException handleHttpResponseError(String apiRequestPath, int code, byte[] returnedContent)
+    private long calculateWaitTimeMillis(ExponentialBackOff backoff, Optional<TDClientException> rootCause)
+    {
+        long waitTimeMillis = backoff.nextWaitTimeMillis();
+        if (rootCause.isPresent() && rootCause.get() instanceof TDClientHttpTooManyRequestsException) {
+            TDClientHttpTooManyRequestsException tooManyRequestsException = (TDClientHttpTooManyRequestsException) rootCause.get();
+            Optional<Long> retryAfterSeconds = tooManyRequestsException.getRetryAfterSeconds();
+            if (retryAfterSeconds.isPresent()) {
+                long maxWaitMillis = config.retryLimit * config.retryMaxIntervalMillis;
```

Comment:
I assume that the user is prepared to have to wait `retryLimit * retryMaxIntervalMillis`.